components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceOrganizationServiceImpl.java

```diff
@@ -0,0 +1,250 @@
+package org.wso2.carbon.device.mgt.jaxrs.service.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.device.mgt.common.*;
+import org.wso2.carbon.device.mgt.core.service.DeviceOrganizationProviderService;
+import org.wso2.carbon.device.mgt.core.service.DeviceOrganizationProviderServiceImpl;
+import org.wso2.carbon.device.mgt.jaxrs.service.api.DeviceOrganizationService;
+
+import javax.validation.Valid;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeviceOrganizationServiceImpl implements DeviceOrganizationService{
+    private static final Log log = LogFactory.getLog(DeviceAgentServiceImpl.class);
+
+    @POST
+    @Path("/add")
+    @Override
+    public Response addDeviceOrganization(@Valid DeviceOrganizationMetadataHolder deviceOrganizationMetadataHolder) {
+        if (deviceOrganizationMetadataHolder == null) {
+            String errorMessage = "The payload of device metadata is empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        String tempId = deviceOrganizationMetadataHolder.getDeviceId();
+        String tempName = deviceOrganizationMetadataHolder.getDeviceName();
+        String tempParent = deviceOrganizationMetadataHolder.getParent();
+        int tempPingMins = deviceOrganizationMetadataHolder.getPingMins();
+        int tempState = deviceOrganizationMetadataHolder.getState();
+        int tempIsGateway = deviceOrganizationMetadataHolder.getIsGateway();
+        boolean result;
+        try {
+            result  = dops.addDeviceOrganization(tempId,tempName,tempParent,tempPingMins,tempState,tempIsGateway);
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while enrolling the device with ID:" + tempId + " to Device Organization ";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GET
+    @Path("/{deviceId}/state")
+    @Override
+    public Response getDeviceOrganizationStateById(@PathParam("deviceId") String deviceId) {
+        if (deviceId.length() <= 0) {
+            String errorMessage = "The parameter of the device organization ID is empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        int result;
```

Comment:
you can define the `result` inside the try-catch block. This will allow further compiler optimizations.

Reply:
The `result` variable here is used after an if conditional block, not a try-catch block.